buildconfig/Jenkins/Conda/conda-buildscript-linux

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash -ex
+
+# This script expects to be in a POSIX environment, it will run our CI workflow on a POSIX environment depending
+# on the flags and args passed. This script will always compile the code, and run the unit tests if a change was
+# made to something other than a .rst file.
+#
+# Script usage:
+# buildscript <path-to-workspace> <cmake-preset-name> [options]
+#
+# Example command to run a PR build on ubuntu:
+# buildscript /jenkins/workspace_dir/ linux-ci --enable-unittests --enable-coverage --enable-docs
+#
+# Expected args:
+#   1. WORKSPACE: path to the workspace/source code that this should run inside
+#   2. CMAKE_PRESET: the CMake preset that should be ran to generate the cmake files for this CI job
+#
+# Possible flags:
+#   --enable-systemtests: Runs the system tests from being compiled or ran
+#   --enable-package: Runs a package being produced
+#   --enable-docs: Runs the docs from being built
+#   --enable-dev-docs: Runs the developer docs from being built
+#   --enable-doctests: Runs the documentation tests
+#   --clean-build: Clears the build folder and builds from scratch
+#   --clean-external-projects: Clear the external projects from the build folder
+#   --use-core-dumps: Activate core dumps (Linux only)
+#   --extra-cmake-flags: Extra flags to pass directly to cmake, enclose in "".
+
+# SCRIPT_DIR discovery from https://stackoverflow.com/a/9107028 by https://stackoverflow.com/users/1184238/andrew-norrie
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+WORKSPACE=$1
+shift
+CMAKE_PRESET=$1
+shift
+EXPECTED_MAMBAFORGE_PATH=$WORKSPACE/mambaforge # Install into the WORKSPACE_DIR
+EXPECTED_CONDA_PATH=$EXPECTED_MAMBAFORGE_PATH/bin/conda
+CONDA_ENV_NAME="mantid-developer"
+CPU_COUNT="$(grep -c ^processor /proc/cpuinfo)" # Not physical but logical cores, includes hyper threaded.
```

Comment:
I think it is worth us using the proc count as a default but also respecting an optional `-j` flag like ninja does? I'm just thinking of a case where a jenkins node is setup and for some reason the owner doesn't want all of the cores used by that process. At the moment you can set `BUILD_THREADS` to control this.

Reply:
I will add a flag to allow this control